src/main/java/com/untamedears/ItemExchange/metadata/PotionMetadata.java

```diff
@@ -3,118 +3,206 @@
+			serialized.append(ExchangeRule.tertiarySpacer);
+			serialized.append(this.base.isExtended() ? 1 : 0);
+			serialized.append(ExchangeRule.tertiarySpacer);
+			serialized.append(this.base.isUpgraded() ? 1 : 0);
+		}
+		serialized.append(ExchangeRule.secondarySpacer);
+		// Serialise addition effects
+		Iterator<PotionEffect> iterator = this.effects.iterator();
 		while(iterator.hasNext()) {
 			PotionEffect effect = iterator.next();
-			
-			serialized.append(effect.getType().getName()).append(ExchangeRule.tertiarySpacer).append(effect.getAmplifier()).append(ExchangeRule.tertiarySpacer).append(effect.getDuration()).append(ExchangeRule.tertiarySpacer).append(effect.isAmbient());
-			
-			if(iterator.hasNext())
+			serialized.append(effect.getType().getId());
+			serialized.append(ExchangeRule.tertiarySpacer);
+			serialized.append(effect.getAmplifier());
+			serialized.append(ExchangeRule.tertiarySpacer);
+			serialized.append(effect.getDuration());
+			serialized.append(ExchangeRule.tertiarySpacer);
+			serialized.append(effect.isAmbient() ? 1 : 0);
+			if (iterator.hasNext()) {
 				serialized.append(ExchangeRule.secondarySpacer);
+			}
 		}
-		
 		return serialized.toString();
 	}
 
-	@Override
-	public boolean matches(ItemStack item) {
-		if(item.hasItemMeta()) {
-			ItemMeta meta = item.getItemMeta();
-			
-			if(meta instanceof PotionMeta) {
-				PotionMeta potionMeta = (PotionMeta) meta;
-				
-				return potionMeta.getCustomEffects().equals(effects);
+	public static PotionMetadata deserialize(String lore) {
+		PotionMetadata metadata = new PotionMetadata();
+		String[] data = lore.split(ExchangeRule.secondarySpacer);
+		for (int i = 0; i < data.length; i++) {
+			// Parse the potion name
+			if (i == 0) {
+				metadata.name = data[i];
+			}
+			// Parse the base effect
+			else if (i == 1) {
+				try {
+					String[] effectData = data[i].split(ExchangeRule.tertiarySpacer);
+					int effectId = Integer.parseInt(effectData[0]);
+					boolean isExtended = Integer.parseInt(effectData[1]) == 1;
+					boolean isUpgraded = Integer.parseInt(effectData[2]) == 1;
+					PotionEffectType effectType = PotionEffectType.getById(effectId);
+					PotionType potionType = PotionType.getByEffect(effectType);
+					metadata.base = new PotionData(potionType, isExtended, isUpgraded);
+				}
+				catch (Exception error) {
+					metadata.base = new PotionData(PotionType.WATER, false, false);
+				}
+			}
+			// Parse any other effect
+			else {
+				try {
+					String[] effectData = data[i].split(ExchangeRule.tertiarySpacer);
+					int effectId = Integer.parseInt(effectData[0]);
+					int amplifier = Integer.parseInt(effectData[1]);
+					int duration = Integer.parseInt(effectData[2]);
+					boolean ambient = Integer.parseInt(effectData[3]) == 1;
+					PotionEffectType type = PotionEffectType.getById(effectId);
+					metadata.effects.add(new PotionEffect(type, duration, amplifier, ambient));
+				}
+				catch (Exception ignored) {}
```

Comment:
Might be good to print and not just silently swallow this exception so actual issues with it can be found and fixed in the future

Reply:
Done :) I also added logging to the other parser